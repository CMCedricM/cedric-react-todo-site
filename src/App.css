*{
  border-radius: 5px;
  color: black; 
  text-align: center;
  background: linear-gradient(to right, #6CD0FF, #1C2E4C);
}
ul{
  list-style-type: none;
  background: none;
}
.todo-item{
  background: none;
  justify-content: center;
  border-style: dotted;
}

.todo-item-cross{
  background: none;
  text-decoration: line-through;
}
.btn{
  background: white;
  margin: 20px;
}
.App{
  background: linear-gradient(to right, #6CD0FF, #1C2E4C);
}

.todo{
  display: flex;
  background: none;
  width: fit-content;
  flex-basis: 0;
  justify-content: center;

}
#mainHeader{
  color: white;
}

.clearButton{
  background: white;
  width: 5rem;
  
}

.todo-input{
  background: white;
  margin-right: 20px;
  width:15rem;
  height: 25px;
  border: none;
  /* border-bottom-right-radius: 0px;
  border-top-right-radius: 0px;
   */
}

/* .todo-button{
  display: none;
} */


.filter-todo{
  height: 27px;
  background: white;
  border: none;
  /* border-top-left-radius: 0px;
  border-bottom-left-radius: 0px; */
}

.input-area{
  display: flex; 
  background: none;
  justify-content: center;
  gap: 20px;
  
}