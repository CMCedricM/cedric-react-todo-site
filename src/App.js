import './App.css'
import ToDoList from './components/ToDoData/ToDoList';
import ToDoForm from './components/toDoForm/ToDoForm';
import UseToDoStore from './stores/zustandToDoStore';


function App() {
  // Functions to Call in zustandStore
  const setToDoText = UseToDoStore((state) => state.setInputText);
  const addToList = UseToDoStore((state) => state.updateToDoData);
  // const clearData = UseToDoStore((state) => state.removeAllItems);
  const clearCurrentViewData = UseToDoStore((state) => state.removeCurrentView);
  const removeAnItem = UseToDoStore((state) => state.removeAnItem )
  const updateCompeletionStatus = UseToDoStore((state) => state.updateCompletionStatus);
  const updateFilterInfo = UseToDoStore((state) => state.updateFilterType); 

  // Functions will handle updating a todo-item 
  const perpareItemUpdate = UseToDoStore((state) => state.prepareForItemUpdate); 
  const setUpdateText= UseToDoStore((state) => state.setUpdateItemText); 
  const submitItemNameUpdate = UseToDoStore((state) => state.updateItemName);
  const itemUpdateAllowed = UseToDoStore((state) => state.updateAllowed);
  const updateText = UseToDoStore((state) => state.updateInputText);
  const filterUpdate= UseToDoStore((state) => state.refreshList);


  // State Variables
  const toDoText = UseToDoStore((state) => state.toDoInputText);
  const filteredData = UseToDoStore((state) => state.filteredItems);
  const filterSettting = UseToDoStore((state) => state.filterType);

  return (
    <div className="App">
      <h1>Welcome!</h1>
      <div className='input-area'>
      <ToDoForm onNewItem={addToList} setToDoText={setToDoText} clearData={clearCurrentViewData} 
      toDoText = {toDoText} setFilter={updateFilterInfo}
      filterSetting={filterSettting}
      triggerListRefresh={filterUpdate}
      ></ToDoForm>
      </div>
      <div className='todo-list-area'>
      <ToDoList toDoItems={filteredData} removeHandler={removeAnItem} completeItemHandler={updateCompeletionStatus}
      updateItemName={perpareItemUpdate} 
      setUpdateText={setUpdateText} 
      finalizeUpdate={submitItemNameUpdate}
      updateAllowed={itemUpdateAllowed}
      updateInputText={updateText}
      refreshList={filterUpdate}
      />
      </div>
    </div>
  );
}

export default App;
