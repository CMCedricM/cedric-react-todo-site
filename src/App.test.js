import { render, screen, within , cleanup} from '@testing-library/react';
import App from './App';
import user from '@testing-library/user-event'
import renderer from "react-test-renderer"


afterEach(cleanup);

// Watch that I didn't accidently delete any important components from the code
it('matches snapshot', () => {
  const tree = renderer.create(<App></App>).toJSON();
  expect(tree).toMatchSnapshot();

})

// Test Whether The Webpage itself loaded or not
it('renders header', () => {
  render(<App />)
  const linkElement = screen.getByText(/Cedric's To Do Site/i);
  expect(linkElement).toBeInTheDocument();
});

// Test whether the boxes can take input
it('adds data to form, changes filters, and then clears', () => {
  render(<App />)
  user.type(getToDoInputBox(), 'Feed Dog');
  const dropdown = getFilterBox(); 
  user.selectOptions(dropdown, within(dropdown).getByRole('option', {name : 'Completed'}));
  expect(getClearButton()).toBeInTheDocument();
  user.click(getClearButton());
});


const getToDoInputBox = () => {
  return screen.getByRole('textbox');
}

const getFilterBox = () => {
  return screen.getByRole('combobox'); 
}

const getClearButton = () => {
  return screen.getByRole('button', {
      name: /clear/i
    });
}