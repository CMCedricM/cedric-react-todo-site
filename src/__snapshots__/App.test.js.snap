// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <h1
    id="mainHeader"
  >
    Welcome!
  </h1>
  <div
    className="input-area"
  >
    <div>
      <div
        className="todo-form"
      >
        <form>
          <input
            className="todo-input"
            data-testid="todo-item-input-field"
            name="toDoInput"
            onChange={[Function]}
            placeholder="Type An Item Here"
            type="text"
            value=""
          />
          <select
            className="filter-todo"
            daa-testid="filter-label"
            label="Filter"
            name="todos"
            onChange={[Function]}
            value="all"
          >
            <option
              value="all"
            >
              Filter: All
            </option>
            <option
              value="completed"
            >
              Completed
            </option>
            <option
              value="incomplete"
            >
              Incomplete
            </option>
          </select>
          <div
            className="action-area "
          >
            <button
              className="btn"
              onClick={[Function]}
            >
              Add Item
            </button>
            <button
              className="btn"
              onClick={[Function]}
            >
              Clear
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="todo-list-area"
  >
    <div
      className="todo-container"
    >
      <ul
        className="todo-list"
      />
    </div>
  </div>
  <div
    className="myFooter-div"
  >
    <footer
      className="myFooter"
    >
      Created by 
      <a
        className="myFooter-name"
        href="https://www.cmcedricm.com/"
        style={
          Object {
            "margin": 1,
          }
        }
      >
        Cedric Men
      </a>
    </footer>
  </div>
</div>
`;
